### PR TITLE
Add onUpdate and onActivate

### LINKED ISSUE #58: Expand Object
Add 2 new fields to ```Object```
```haskell
onUpdate :: Game -> Game
onActivate :: Game -> Game
``` 
```onUpdate``` is called on every update for every object
```onActivate``` is called when object is activated

### PR DESCRIPTION
**Pull request by dm-gh**
Resolves #58

### CODE REVIEW

File: src/Util/Common.hs [hunk 1]
```diff
@@ -49,4 +49,13 @@ updateCamera game = game
...
+                    ]
+
+
+gameObjects :: Game -> [Object]
+gameObjects game = map playerObject (gamePlayers game)
+                   ++ levelObjects (gameLevel game)
+
+
+updateObjects :: Game -> Game
+updateObjects game = foldr (\obj -> (objectOnUpdate obj) obj) game (gameObjects game)
```
Comment (alois9866) [inline src/Util/Common.hs]: Consider changing (objectOnUpdate obj) to objectOnUpdate obj $ ...

File: src/Straminata.hs [hunk 2]
```diff
@@ -82,8 +86,40 @@ player2InitialState = Player
...
--- advanceGame time = updatePlayers . updateCamera . moveObjects time . movePlayers time
-advanceGame time = updateCamera . moveObjects time . movePlayers time
+advanceGame time = updateCamera . moveObjects time . movePlayers time . updateObjects
+
+
+activatePlayer :: String -> Object -> Game -> Game
+activatePlayer name object game =
+  foldr (\player -> (objectOnActivate . playerObject $ player)
+                    (objectsCollide object (playerObject player))
+                    (playerObject player)
```
Comment (alois9866) [inline src/Straminata.hs]: Don't forget that name is unique and you do not really need to iterate over all players.

File: src/Straminata.hs [hunk 3]
```diff
@@ -82,8 +86,40 @@ player2InitialState = Player
...
+        )
+        game
+        (filter isTarget (gamePlayers game))
+  where isTarget player = (objectName . playerObject $ player) == name
+
+
+resizeSelf :: Bool -> Object -> Game -> Game
+resizeSelf state self game = game
```
Comment (alois9866) [inline src/Straminata.hs]: Cool and good.

File: src/Straminata.hs [hunk 4]
```diff
@@ -82,8 +86,40 @@ player2InitialState = Player
...
+                                       then enlarge player
+                                       else reduce player
+                                  else player
+                      ) (gamePlayers game)
+  }
+  where isSelf player = (objectName . playerObject $ player) == (objectName self)
+        enlarge = changeSize (Position (-30, -40), Position (90, 120))
+        reduce = changeSize (Position (0, 0), Position (60, 80))
+        changeSize rect player = player
+          { playerObject = (playerObject player)
```
Comment (alois9866) [inline src/Straminata.hs]: Consider scaling hitbox too.

File: src/Straminata.hs (hunk 2 above)
Comment (dm-gh) [reply src/Straminata.hs]: i'm not iterating over all players i'm iterating over 1 player. I like this version better

File: src/Straminata.hs (hunk 4 above)
Comment (dm-gh) [reply src/Straminata.hs]: done

File: src/Straminata.hs (hunk 3 above)
Comment (dm-gh) [reply src/Straminata.hs]: ok

File: src/Util/Common.hs (hunk 1 above)
Comment (dm-gh) [reply src/Util/Common.hs]: done

**Pull request merged**